staging/src/k8s.io/client-go/scale/client.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scale
+
+import (
+	"sync"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/api/meta"
+	scaleapi "k8s.io/client-go/pkg/apis/autoscaling/v1"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/pkg/api"
+)
+
+// scaleClient is similar to dynamic.ClientPool, except that it knows a bit
+// more about what it's doing, so it can be more specific.
+type scaleClient struct {
```

Comment:
@DirectXMan12 https://github.com/kubernetes/kubernetes/issues/36331 proposed an alternative solution. In short, it was suggesting that we add a method to restclient.Requst to allow setting the groupVersion segment of the url dynamically. It might be easier than writing a scaleclient. Also, I think there might be subresources other than `scale` that needs to operate across the group boundary, #36331 can provide a generic solution.